rosidl_generator_c/src/string_functions.c

```diff
@@ -24,9 +24,10 @@ rosidl_generator_c__String__init(rosidl_generator_c__String * str)
   if (!str) {
     return;
   }
-  str->data = NULL;
+  str->data = malloc(1);
```

Comment:
OK, since everyone is complaining about error checking in this class, I will change all of the function signatures to return bool, and return false on an error state.